src/Deinterleave.cpp

```diff
@@ -419,12 +435,12 @@ Expr extract_odd_lanes(const Expr &e, const Scope<> &lets) {
 
 Expr extract_even_lanes(const Expr &e, const Scope<> &lets) {
     internal_assert(e.type().lanes() % 2 == 0);
-    return deinterleave(e, 0, 2, (e.type().lanes() + 1) / 2, lets);
+    return deinterleave(e, 0, 2, e.type().lanes() / 2, lets);
```

Comment:
If the starting lane is zero, and there are 3 lanes total, shouldn't there be two lanes in the result?

Reply:
Oh, there's an assert that the number of lanes is even, never mind.